In [34]:
import pandas as pd
import matplotlib.pyplot as plt

In [35]:
outlets = pd.read_csv("Data/QSR_outlet_dataset_cleaned.csv", keep_default_na= False)
socioeconomic = pd.read_csv("Data/QSR_socioeconomic_dataset_cleaned.csv")

In [36]:
print(f"QSR outlet row and column: {outlets.shape}")
print(f"Socio-Economic row and column: {socioeconomic.shape}")

QSR outlet row and column: (2823, 17)
Socio-Economic row and column: (20, 9)


In [37]:
outlets.head()

,Outlet ID,Brand,Region,Province,Province_Shorthand,City,Menu_Category,Price_Range,Customer_Rating,Review_Count,Latitude,Longitude,Location_Type,Address,Source,Price_Range_Cleaned,Price_Range_Imputed
0,1,McDonald's,Valle d'Aosta,Aosta,AO,Aosta,Burgers,EUR 1-10,3.8,3974,45.739527,7.345432,Commercial Area,"Aosta, 11100 Aosta, Aosta Valley",https://www.mcdonalds.it/elenco-ristoranti,EUR 1-10,False
1,2,McDonald's,Liguria,Genova,GE,Genova,Burgers,EUR 1-10,3.7,2244,44.406840,8.936673,City Center,Via Venti Settembre 1,https://www.mcdonalds.it/elenco-ristoranti,EUR 1-10,False
2,3,McDonald's,Liguria,Genova,GE,Genova,Burgers,EUR 1-10,3.9,2170,44.413723,8.879941,Shopping Mall,"Via Paolo Mantovani, 17, 16149 Genova GE",https://www.mcdonalds.it/elenco-ristoranti,EUR 1-10,False
3,4,McDonald's,Liguria,Genova,GE,Genova,Burgers,EUR 1-10,2.9,303,44.417238,8.921427,Transport Hub,"Piazza Acquaverde, 16134 Genova GE",https://www.mcdonalds.it/elenco-ristoranti,EUR 1-10,False
4,5,McDonald's,Liguria,Genova,GE,Genova,Burgers,EUR 1-10,3.8,3290,44.411409,8.928781,Transport Hub,"Via di Sottoripa, 127R/133, 16124 Genova GE",https://www.mcdonalds.it/elenco-ristoranti,EUR 1-10,False


In [38]:
socioeconomic.head()

,Region,Province_Count,Region_Area,Population,Population Density,Disposable Income per capita,Total Tourism Flow,Foreign Tourism Flow,Median Income
0,Lombardia,12,23863.65,10065694,421.8,27243,41795073.0,26300388.0,32485
1,Valle d'Aosta,1,3260.90,122554,37.6,25152,3692878.0,1439669.0,32900
2,Piemonte,8,25387.07,4255006,167.6,24416,14410448.0,7180603.0,30267
3,Trentino-Alto Adige,2,13605.50,1090818,80.2,28130,55227131.0,33544804.0,37563
4,Veneto,7,18407.42,4857460,263.9,24103,71896863.0,49799037.0,32272


In [39]:
#Checking Data Types

outlets.info()

<class 'pandas.DataFrame'>
RangeIndex: 2823 entries, 0 to 2822
Data columns (total 17 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Outlet ID            2823 non-null   int64  
 1   Brand                2823 non-null   str    
 2   Region               2823 non-null   str    
 3   Province             2823 non-null   str    
 4   Province_Shorthand   2823 non-null   str    
 5   City                 2823 non-null   str    
 6   Menu_Category        2823 non-null   str    
 7   Price_Range          2823 non-null   str    
 8   Customer_Rating      2823 non-null   float64
 9   Review_Count         2823 non-null   int64  
 10  Latitude             2823 non-null   float64
 11  Longitude            2823 non-null   float64
 12  Location_Type        2823 non-null   str    
 13  Address              2823 non-null   str    
 14  Source               2823 non-null   str    
 15  Price_Range_Cleaned  2823 non-null   str    
 16 

In [ ]:
#Checking Data Types

socioeconomic.info()

<class 'pandas.DataFrame'>
RangeIndex: 20 entries, 0 to 19
Data columns (total 9 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Region                        20 non-null     str    
 1   Province_Count                20 non-null     int64  
 2   Region_Area                   20 non-null     float64
 3   Population                    20 non-null     int64  
 4   Population Density            20 non-null     float64
 5   Disposable Income per capita  20 non-null     int64  
 6   Total Tourism Flow            20 non-null     float64
 7   Foreign Tourism Flow          20 non-null     float64
 8   Median Income                 20 non-null     int64  
dtypes: float64(4), int64(4), str(1)
memory usage: 1.5 KB


In [41]:
# Checking missing vlaues for QSR outlet

(outlets == "").sum()

Outlet ID               0
Brand                   0
Region                  0
Province                0
Province_Shorthand      0
City                    0
Menu_Category           0
Price_Range            64
Customer_Rating         0
Review_Count            0
Latitude                0
Longitude               0
Location_Type           0
Address                 0
Source                  0
Price_Range_Cleaned     0
Price_Range_Imputed     0
dtype: int64

In [42]:
#Checking for missing values

socioeconomic.isna().sum()

Region                          0
Province_Count                  0
Region_Area                     0
Population                      0
Population Density              0
Disposable Income per capita    0
Total Tourism Flow              0
Foreign Tourism Flow            0
Median Income                   0
dtype: int64

In [ ]:
# Summary statistics for the numeric variables of the QSR outlet dataset
outlets.describe()

,Outlet ID,Customer_Rating,Review_Count,Latitude,Longitude
count,2823.00000,2823.000000,2823.000000,2823.000000,2823.000000
mean,1412.00000,3.869501,1857.577046,43.796181,11.152527
std,815.07423,0.413796,2966.380768,2.164981,2.231580
min,1.00000,0.000000,0.000000,36.901523,6.892710
25%,706.50000,3.700000,246.500000,41.916067,9.204082
50%,1412.00000,3.900000,1036.000000,44.798913,11.016003
75%,2117.50000,4.100000,2634.500000,45.491114,12.542852
max,2823.00000,5.000000,59200.000000,46.679271,18.197403


From this data we can know:
    <br />1, The average customer rating is 3.869501 out of 5 for all QSRs in Italy. which is below 4.
    <br>2, The average outlet has about 1858 reviews. But the review counts are very spread out. So, the median would be a better measure of a typical outlet
    <br>3, The coordinates are inside Italy. As Italy has Latitude between (35.5, 47.1) and Longitude between (6.6, 18.6)

In [45]:
# Summary statistics for numeric socioeconomic variables

socioeconomic.describe()

,Province_Count,Region_Area,Population,Population Density,Disposable Income per capita,Total Tourism Flow,Foreign Tourism Flow,Median Income
count,20.000000,20.000000,2.000000e+01,20.000000,20.000000,2.000000e+01,2.000000e+01,20.000000
mean,5.350000,15103.642000,2.947141e+06,177.885000,21767.100000,2.235850e+07,1.170911e+07,29163.300000
std,3.013566,7448.375038,2.509338e+06,112.313177,3843.639132,2.019318e+07,1.369814e+07,4658.277297
min,1.000000,3260.900000,1.225540e+05,37.600000,16173.000000,4.947860e+05,4.749700e+04,22225.000000
25%,3.500000,9167.117500,1.167826e+06,95.425000,17993.000000,7.776650e+06,1.668786e+06,24696.250000
50%,5.000000,14446.425000,1.691030e+06,158.300000,22732.000000,1.524733e+07,6.999274e+06,29490.500000
75%,7.250000,22586.345000,4.551555e+06,215.525000,24583.250000,3.983087e+07,1.463992e+07,32992.000000
max,12.000000,25832.390000,1.006569e+07,421.800000,28130.000000,7.189686e+07,4.979904e+07,37563.000000


From this result we can understand:
<br>1, How large Italian regions are in population. Italian regions are very different in population size. The average region has 2.95 million people but the median has only 1.69 million population size. Meaning, very few large regions are pulling the average up. Therefore, Most Italian regions are relatively small or medium-sized, while a few large regions contain a very high share of the national population.

2, How different are regions is population density. they vary very much in how crowded they are. some regions have low density, around 37.6 people per km2 while others have 442 people per km2. Therefore, QSR outlets may be more concentrated in dense regions because dense areas have more foot traffic, more urban centers, and more commercial activity.

3, There is also a clear income difference among the regions. Higher-income regions may support more QSR outlets, especially brands with higher price ranges, but this needs to be tested later with correlation and regression.

4, The maximum tourism flow is huge compared to the minimum. Some regions receive less than 500,000 tourist flows, while the highest region receives about 71.9 million. Telling us that, tourism is concentrated in small number of regions

5, Median income also varies strongly across regions. Some regions have much stronger household income conditions than others, which may influence restaurant demand, brand location strategy, and price-range distribution.

In [48]:
# Count outlets by QSR type

outlets["Brand"].value_counts()


Brand
McDonald's              790
La Piadineria           528
Autogrill               282
Old Wild West           243
Alice Pizza             181
Burger King             177
Roadhouse Restaurant    168
KFC                     153
Chef Express             99
Billy Tacos              69
Starbucks                48
Spontini                 25
Spizzico                 16
Popeyes                  15
Five Guys                10
Burgez                   10
Subway                    9
Name: count, dtype: int64